code/datums/alert.dm

```diff
@@ -0,0 +1,181 @@
+///Represents a single source of alerts, one alert handler will only ever count for one alert per listener
+/datum/alert_handler
+	///A list of alert type -> list of area names we currently have alerts in
+	var/list/sent_alerts = list()
+	///Our source atom
+	var/atom/source_atom
+
+/datum/alert_handler/New(atom/_source_atom)
+	if(istype(_source_atom))
+		source_atom = _source_atom
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#when-passing-vars-through-new-or-initializes-arguments-use-srcvar

Reply:
Oh no I forgot the guideline. jesus